.github/workflows/pypi-cd.yml

```diff
@@ -7,67 +7,62 @@ on:
     runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: false
       matrix:
-        os: [ubuntu-latest, macOS-latest, windows-latest]
-        python-version: ["3.7", "3.8", "3.9", "3.10"]
-        node-version: ["14.x"]
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        arch: [auto]
         include:
-          - { os: ubuntu-latest, python-version: "3.7", python-abis: "cp37-cp37m" }
-          - { os: ubuntu-latest, python-version: "3.8", python-abis: "cp38-cp38" }
-          - { os: ubuntu-latest, python-version: "3.9", python-abis: "cp39-cp39" }
-          - { os: ubuntu-latest, python-version: "3.10", python-abis: "cp310-cp310" }
-          - { os: windows-latest, python-version: "3.10", build-static: 1 }
+          - os: ubuntu-latest
+            arch: aarch64
+          - os: macos-latest
+            arch: universal2
 
     steps:
       - name: Check out code
         uses: actions/checkout@v2
 
-      - name: Use Node.js ${{ matrix.node-version }}
-        uses: actions/setup-node@v2
-        with:
-          node-version: ${{ matrix.node-version }}
+      - name: Set up QEMU
+        if: ${{ matrix.arch == 'aarch64' }}
+        uses: docker/setup-qemu-action@v1
 
-      - name: Set up conda ${{ matrix.python-version }}
+      - name: Build wheels
+        uses: pypa/cibuildwheel@v2.3.1
         env:
-          PYTHON: ${{ matrix.python-version }}
-        shell: bash
-        run: |
-          source ./ci/install-conda.sh
-          python -m pip install --upgrade pip setuptools wheel coverage;
+          CIBW_BEFORE_BUILD: pip install -r ci/requirements-wheel.txt
+          CIBW_ARCHS: ${{ matrix.arch }}
+          CIBW_SKIP: pp* *-musllinux* cp36-* cp310-win32 *i686
 
-      - name: Install dependencies
-        env:
-          WITH_HADOOP: ${{ matrix.with-hadoop }}
-          WITH_KUBERNETES: ${{ matrix.with-kubernetes }}
-          NO_COMMON_TESTS: ${{ matrix.no-common-tests }}
-        shell: bash
+      - name: Build source
+        if: ${{ matrix.os == 'wnidows-latest' }}
         run: |
-          source ./ci/reload-env.sh
-          export DEFAULT_VENV=$VIRTUAL_ENV
+          python -m pip install numpy scipy cython
+          python setup.py sdist --dist-dir=wheelhouse
 
-          if [[ "$PYTHON" =~ "3.9" ]]; then
-            conda install -n test --quiet --yes -c conda-forge python=$PYTHON pyarrow
-          fi
-
-          pip install numpy scipy cython
-          pip install -e ".[dev,extra]"
-          conda list -n test
-
-      - name: Deploy packages
-        if: startsWith(github.ref, 'refs/tags/') && matrix.no-deploy != '1'
+      - name: Release to pypi
         shell: bash
         env:
-          PYABI: ${{ matrix.python-abis }}
-          BUILD_STATIC: ${{ matrix.build-static }}
-          PYPI_PWD: ${{ secrets.PYPI_PASSWORD }}
+          TWINE_USERNAME: __token__
+          TWINE_PASSWORD: ${{ secrets.pypi_password }}
```

Comment:
Keep `PYPI_PWD: ${{ secrets.PYPI_PASSWORD }}` in existing pipeline to make script work.